camply/notifications/twilio.py

```diff
@@ -5,12 +5,12 @@
 import logging
 from typing import List
 
+from twilio.rest import Client
```

Comment:
Hmm... I forgot how confusing extras can be. maybe move the `from twilio.rest import Client` into the `__init__` and add an error command requesting the user to install camply with the Twilio extra.

Also add `twilio` to the `dev` extras as well which get installed inside tox

Reply:
```python
class TwilioNotifications(BaseNotifications):
    """
    Push Notifications via Twilio
    """

    def __init__(self):
        try:
            from twilio.rest import Client
        except ImportError:
           raise RuntimeError("Looks like `twilio` isn't installed. Install it with `pip install camply[twilio]`")
        self.client = Client(TwilioConfig.ACCOUNT_SID, TwilioConfig.AUTH_TOKEN)
        ...
```

Reply:
Thanks for the pointers! Python is not my primary language (by far), so your pointers are very helpful.

Since `send_campsites()` and `send_message()` are static/class methods, I stored the initialized `Client` (and `phone_nums`) as class variables. It all seems to work (I was able to run and get SMS), but I admit I haven't thought through if this is the 'best' way to do it. Happy to hear more feedback on how to structure this.

Reply:
Ahh - yeah this definitely works. I think the _best_ way to do this is to actually make the method an instance method. AFAIK the `BaseNotification` classes are always instantiated anyways so their key functions don't need to be `classmethods` anyways. 

What you did absolutely works - but only because we're instantiating the `TwilioNotifications` class anyways. Take this example:

```python
import requests


class ExamplePythonClass:
    """
    Example Class to Explain Juftin's Dumb Use of classmethods
    """

    def __init__(self):
        ExamplePythonClass.session = requests.Session()

    @classmethod
    def do_a_thing(cls) -> requests.Response:
        response = cls.session.get("https://www.google.com/")
        print(response.status_code)
        return response
```

With the above code, this would work:
```python
resp = ExamplePythonClass().do_a_thing()
```

But calling the function like a normal class/static method raises an error: `AttributeError: type object 'ExamplePythonClass' has no attribute 'session'`:
```python
resp = ExamplePythonClass.do_a_thing()
```

All of this is to say - what you've done here works and is good to go. It's my insistence that those functions be classmethods that made this so confusing. I need to go through the Notification providers myself and do some cleanup after your code is merged.

Reply:
... okay I just signed up for a Twilio account. I'm so stoked for this. 

One other thing I'm noticing - the `twilio` package has really noisy logging. I'm wondering if we should suppress it:
```
[2022-09-24 00:03:50] INFO     -- BEGIN Twilio API Request --
[2022-09-24 00:03:50] INFO     POST Request: https://api.twilio.com/2010-04-01/Accounts/xxxxxxxxxxxxxxxxxxxxxxxxxxxx/Messages.json
[2022-09-24 00:03:50] INFO     Headers:
[2022-09-24 00:03:50] INFO     User-Agent : twilio-python/7.14.1 (Darwin x86_64) Python/3.9.5
[2022-09-24 00:03:50] INFO     X-Twilio-Client : python-7.14.1
[2022-09-24 00:03:50] INFO     Accept-Charset : utf-8
[2022-09-24 00:03:50] INFO     Content-Type : application/x-www-form-urlencoded
[2022-09-24 00:03:50] INFO     Accept : application/json
[2022-09-24 00:03:50] INFO     -- END Twilio API Request --
[2022-09-24 00:03:50] DEBUG    Starting new HTTPS connection (1): api.twilio.com:443
[2022-09-24 00:03:50] DEBUG    https://api.twilio.com:443 "POST /2010-04-01/Accounts/xxxxxxxxxxxxxxxxxxxxxxxxxxxx/Messages.json HTTP/1.1" 201 1363
[2022-09-24 00:03:50] INFO     Response Status Code: 201
[2022-09-24 00:03:50] INFO     Response Headers: {'Date': 'Sat, 24 Sep 2022 06:03:50 GMT', 'Content-Type': 'application/json; charset=utf-8', 'Content-Length': '1363', 'Connection': 'keep-alive',
                               'Twilio-Concurrent-Requests': '1', 'Twilio-Request-Id': 'xxxxxxxxxxxxxxxxxxxxxxxxxxxx', 'Twilio-Request-Duration': '0.128', 'Access-Control-Allow-Origin': '*',
                               'Access-Control-Allow-Headers': 'Accept, Authorization, Content-Type, If-Match, If-Modified-Since, If-None-Match, If-Unmodified-Since', 'Access-Control-Allow-Methods': 'GET,
                               POST, DELETE, OPTIONS', 'Access-Control-Expose-Headers': 'ETag', 'Access-Control-Allow-Credentials': 'true', 'X-Powered-By': 'AT-5000', 'X-Shenanigans': 'none', 'X-Home-Region':
                               'us1', 'X-API-Domain': 'api.twilio.com', 'Strict-Transport-Security': 'max-age=31536000'}
```

Reply:
Thanks for the explanation. I agree that we should just use instance methods instead of class/static methods.

And also agreed on the noisiness of the Twilio logs. Added the change to logging level.